src/components/token/Tokens.js

```diff
@@ -24,18 +27,101 @@ class Tokens extends React.Component {
             tokens: [],
             hasAfter: false,
             hasBefore: false,
-            searchText: ""
+            searchText: "",
+            sortBy: "uid",
+            order: "asc",
+            page: 1,
+            //This variable stores the pages and its searchAfter params that is passed to explorer-service
+            pageSearchAfter: [],
+            loading: true
         }
     }
 
-    onSearchButtonClicked = () => {
-        // TODO: Implement behavior when search button is clicked
+    componentDidMount = async() => {
+        //"Click" on search to make the first query
+        await this.onSearchButtonClicked();
+        this.setState({loading: false})
+    }
+
+    resetPageSearchAfter = () => {
+        this.setState({pageSearchAfter: [
+            {
+                page: 1,
+                searchAfter: []
+            }
+        ]})
+    }
+
+    getTokens = async (searchAfter) => {
+        const tokensRequest = await tokensApi.getList(this.state.searchText, this.state.sortBy, this.state.order, searchAfter)
+        let tokens = get(tokensRequest, 'data', {hits:[], 'has_next': false})
+        forEach(tokens.hits, (token) => {
+            if(isNil(token.uid)) {
+                token.uid = token.id
+            }
+            token.type = token.nft ? 'NFT' : 'Custom Token'
+        })
+
+        return tokens;
+    }
+
+    onSearchButtonClicked = async () => {
+        //When search button is clicked, results return to the first page
+        this.resetPageSearchAfter();
+        const tokens = await this.getTokens([]);
+
+        this.setState({page: 1, tokens: tokens.hits, hasAfter: tokens.has_next, hasBefore: false})
     }
 
     onSearchTextChanged = (event) => {
         this.setState({ searchText: event.target.value })
     }
 
+    onSearchTextKeyPressed = (event) => {
+        if(event.charCode === 13) {
+            this.onSearchButtonClicked();
+        }
+    }
+
+    nextPageClicked = async (event) => {
+        const nextPage = this.state.page+1
+        let searchAfter = get(find(this.state.pageSearchAfter, {page: nextPage}), 'searchAfter', [])
+
+        //Calculate searchAfter of next page if not already calculated
+        if(isEmpty(searchAfter)) {
+            const lastCurrentTokenSort = get(last(this.state.tokens), 'sort', []);
```

Comment:
What's the expected format of elements in `tokens` array?

Reply:
Each `tokens` element must be an object with `uid,` `name,` `symbol,`  `nft`, and `sort`.

`sort` is an array with two strings. The value is given by ElasticSearch and it is passed back when we want to change page.

Reply:
I understand how `sort` and `searchAfter` work on the ES query context, but should we rename on our api return so it doesn't get confusing?

Reply:
I think all those fields and explanation should be added to the state docstring. For me this would be enough to be honest.

Reply:
I've updated the state docstring with this explanation.

Reply:
>         * tokens: List of tokens currently being rendered

This should be improved in my opinion. You should write what you said in the comment so nobody will have questions anymore.

> Each tokens element must be an object with uid, name, symbol, nft, and sort.
> sort is an array with two strings. The value is given by ElasticSearch and it is passed back when we want to change page.